packages/ui-components/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {useState, useEffect} from 'react';
+import styled from 'styled-components';
+import {IconChevronRight, IconChevronLeft} from '../icons/interface';
+import {ButtonText, ButtonIcon} from '../button';
+export interface PaginationProps {
+  /**
+   * white background
+   */
+  bgWhite?: boolean;
+  /**
+   * Number of total steps
+   */
+  totalSteps?: number;
+  /**
+   * active steps
+   */
+  defaultStep?: number;
+  onChange?: (step: number) => void;
+}
+
+/**
+ * Default UI component
+ */
+export const Pagination: React.FC<PaginationProps> = ({
+  totalSteps = 10,
+  defaultStep = 1,
+  bgWhite = false,
+  onChange,
+}) => {
+  const [step, setStep] = useState<number>(defaultStep);
+
+  useEffect(() => {
+    onChange && onChange(step);
+  }, [onChange, step]);
+
+  function ButtonList() {
+    const List = [];
+    const Distance = 3;
+
+    if (totalSteps <= 7) {
+      for (let i = 1; i <= totalSteps; i++) {
+        List.push(
+          <ButtonText
+            mode="secondary"
+            size="large"
+            isActive={step === i}
+            onClick={() => setStep(i)}
+            {...(bgWhite && {bgWhite})}
+            label={`${i}`}
+          />
+        );
+      }
+      return <>{List}</>;

```

Comment:
```suggestion
      return List;
```

Maybe a bit less complicatedd :p

Reply:
the return of the arrey is a react element and can not return an array. So I tried to return an array as react element